shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -619,6 +630,21 @@ - (UITextRange*)rangeEnclosingPosition:(UITextPosition*)position
 }
 
 - (UITextRange*)lineEnclosingPosition:(UITextPosition*)position {
+  // TODO(hellohuanlin): Remove iOS 17 check. The same logic should apply to older versions too.
+  if (@available(iOS 17.0, *)) {
+    // The end of document (with backward affinity) should not be part of any line, since end is
```

Comment:
maybe mention this is iOS expectation, since in flutter (text.length, TextAffinity.upstream) is still considered pointing to the last character due to gesture reason. e.g., we need to tell the difference between user hitting the last half of the last character vs outside of the last character to the right.